main/lsp/LSPPreprocessor.cc

```diff
@@ -0,0 +1,395 @@
+#include "main/lsp/LSPPreprocessor.h"
+#include "absl/strings/match.h"
+#include "absl/strings/str_replace.h"
+#include "main/lsp/lsp.h"
+#include "main/pipeline/pipeline.h"
+
+using namespace std;
+
+namespace sorbet::realmain::lsp {
+
+constexpr int INITIAL_VERSION = 0;
+
+LSPPreprocessor::LSPPreprocessor(unique_ptr<core::GlobalState> initialGS, LSPConfiguration config, WorkerPool &workers,
+                                 const std::shared_ptr<spdlog::logger> &logger)
+    : ttgs(TimeTravelingGlobalState(config, logger, workers, move(initialGS), INITIAL_VERSION)), config(move(config)),
+      workers(workers), logger(logger), nextMessageId(INITIAL_VERSION + 1) {
+    const auto &gs = ttgs.getGlobalState();
+    finalGSErrorQueue = make_shared<core::ErrorQueue>(gs.errorQueue->logger, gs.errorQueue->tracer);
+    // Required for diagnostics to work.
+    finalGSErrorQueue->ignoreFlushes = true;
```

Comment:
we should probably give `ignoreFlushes` a better name... though I don't have a better one...
Maybe make the error queue take a boolean if it's an LSP queue at creation?

Reply:
That works fine for me. `ignoreFlushes` is *accurate*, since it ignores the method calls contained within the typechecking routines that flush errors to the logger.

Reply:
Your call, I just find that I get confused my the way I named this variable, if you like it as is: please feel free to keep :-)